fhir-server/src/main/java/au/csiro/pathling/fhirpath/NonLiteralPath.java

```diff
@@ -72,8 +87,29 @@ protected NonLiteralPath(@Nonnull final String expression, @Nonnull final Datase
     this.dataset = applySelection(hashedDataset, this.idColumn);
   }
 
+  /**
+   * Returns the specified child of this path, if there is one.
+   *
+   * @param name The name of the child element
+   * @return an {@link ElementDefinition} object
+   */
   @Nonnull
-  @Override
   public abstract Optional<ElementDefinition> getChildElement(@Nonnull final String name);
 
+  /**
+   * Gets a this {@link Column} from any of the inputs, if there is one.
+   *
+   * @param inputs a collection of objects
+   * @return a {@link Column}, if one was found
+   */
+  @Nonnull
+  public static Optional<Column> findThisColumn(@Nonnull final Object... inputs) {
```

Comment:
Seems to be very similar to `FhirPath.findIdColumn`. Perhaps both can be refactored to use a helper method.

Reply:
I'm going to hold off on doing this until #151 is fixed - as I have a feeling that `findIdColumn` is wrongly implemented anyway...